backend/de.metas.business/src/main/java/de/metas/project/budget/BudgetProjectRepository.java

```diff
@@ -69,7 +87,8 @@ public List<BudgetProject> queryAllActiveProjects(@NonNull final InSetPredicate<
 				.collect(ImmutableList.toImmutableList());
 	}
 
-	public static Optional<BudgetProject> fromRecord(final I_C_Project record)
+	@NonNull
+	public Optional<BudgetProject> fromRecord(final I_C_Project record)
```

Comment:
usually, if you cannot make this method a pure function (basically make it static, don't use external services, no side effects), it's a sign of bad design.